src/RapidezServiceProvider.php

```diff
@@ -291,6 +294,19 @@ function ($filename) use ($manifest) {
         return $this;
     }
 
+    protected function bootFilters(): self
+    {
+        $productSynonymFields = Eventy::filter('index.products.synonym-fields', ['name']);
+        Eventy::addFilter('index.products.settings', fn ($filters) => array_merge($filters ?: [], [WithSynonyms::class]));
+        Eventy::addFilter('index.products.mapping', fn ($filters) => array_merge($filters ?: [], [[WithSynonyms::class, 'fields' => $productSynonymFields]]));
+
+        $categorySynonymFields = Eventy::filter('index.categories.synonym-fields', ['name']);
+        Eventy::addFilter('index.categories.settings', fn ($filters) => array_merge($filters ?: [], [WithSynonyms::class]));
+        Eventy::addFilter('index.categories.mapping', fn ($filters) => array_merge($filters ?: [], [[WithSynonyms::class, 'fields' => $categorySynonymFields]]));
+
+        return $this;
+    }
```

Comment:
Do we also want to use these events from the core? Maybe cleaner to just have them and able to use the filters to override/extend it?